lib/test-lib-installations.c

```diff
@@ -0,0 +1,147 @@
+#include "config.h"
+
+#include <flatpak.h>
+#include <glib.h>
+
+static void
+test_get_system_installations (void)
+{
+  g_autoptr (GPtrArray) installs = NULL;
+  g_autoptr(GError) error = NULL;
+  int i;
+
+  installs = flatpak_get_system_installations (NULL, &error);
+  if (installs == NULL)
+    {
+      g_print ("Error getting system installations: %s", error->message);
+      g_assert (FALSE);
+      return;
+    }
+
+  g_print ("\nInstallations found: %d\n", installs->len);
+  for (i = 0; i < installs->len; i++)
+    {
+      FlatpakInstallation *installation = (FlatpakInstallation*) g_ptr_array_index (installs, i);
+      g_autoptr(GFile) installation_path = flatpak_installation_get_path (installation);
+
+      g_autofree char *path = g_file_get_path (installation_path);
+      g_print ("\nInstallation found: %s\n", path);
+    }
+
+  g_assert (TRUE);
+}
+
+static void
+test_new_system_with_id (void)
+{
+  const char *installs[] = { "endless-games", "endless-sdcard", NULL };
```

Comment:
Definitely. As mentioned in the PR this was a sloppy way for me to test what I was doing without worrying much about how to write this in a generic way (I still need to figure that out) before getting any feedback on this PR.

But yes, this commit is not mergeable as it is at all. Will work on a better version on it.